### PR TITLE
Using Relative paths

### PR DESCRIPTION
**Pull request by plribeiro3000**
relates to fog/fog#3430 

This is a huge diff but sounds like it did the trick. I'm opening this as a start point so we can discuss about it and find the best approach to tackle it.

The changes are:

* Use `autoload` so extra files only get loaded when needed
* Use relative paths
* Move the logic in the `fog/brightbox/core` file to the `fog/brightbox` since this is the only one bundler will load automatically.
* Remove some unnecessary requires in the models
* Move some files so the path will match the namespace

### CODE REVIEW

**Comment by tokengeek** [comment]
@plribeiro3000 My biggest problem with these changes are making the namespacing worse, while we're talking about fixing it for everyone to be idiomatic.

All the code should (eventually) be within `Fog::Brightbox` namespace and this is a breaking change in a few places. Then there will be another breaking change correcting them.

So this seems to be a step beyond autoloading at the moment.

**Comment by tokengeek** [comment]
This will immediately break - https://github.com/fog/fog/blob/master/tests/brightbox/compute/image_selector_tests.rb#L48

However it's value as part of the main `fog` tests has long passed. So I'll remove those support tests.

**Comment by plribeiro3000** [comment]
@tokengeek Could you give this another look?

I did remove the namespace changes. Also, i've tried to load as the minimum as possible of constants in the main file so the loading time of `fog-brightbox` will be faster.

LMK of your concerns.

**Comment by tokengeek** [comment]
@plribeiro3000 I finally took some time to look at this again. Sorry it's taken so long.

There's still many, many file renames which breaks some of our internal applications I'm not convinced about.

Since the original scope was switching to relative paths, I'm not sure why the file renames are needed at this stage.

Referencing `File.expand_path("../brightbox/compute", __FILE__)` verses `File.expand_path("../compute/brightbox", __FILE__)` is unlikely to give a performance boost when loading.

We can autoload from the original file locations if say `Fog::Compute::Brightbox` is referenced.

As I mentioned above, since we should be aiming for `Fog::Provider` namespace the files are already in the correct place under `fog/brightbox` - moving all the files to the correct location for the current namespace, then moving them all back again seems like needless churn.

I've also tried to time the differences, by requiring `fog` with bundler pointing at the two versions of the code base. Can't see anything statistically significant.

In short if we're no longer seeing load times as bad as before, is it worth either of us spending our time making cleaning up this PR?

**Comment by plribeiro3000** [comment]
@tokengeek Yeah. You are right. This was supposed to be just a relative path PR but it turn out to be a lot more.

All the files moving was to reference their namespace in a good way. There are some classes like `Fog::Compute::Brightbox` defined in a file `fog/brightbox/compute` which is kind of confusing for starters with with the codebase.

As you pointed out, we will be moving this namespace again so there is no need to work twice on this. I will close this PR and send a short one just with the relative stuff.

There is no significative performance issues when you look at the time but it might be when you look at the amount of loaded objects. We need to use relative paths and `autoload` to prevent bundler to autoload the entire library from the beginning. The providers inside `fog` are not a problem because of the fog lazy loading but the ones extracted need to use autoload to prevent this since bundler will load [this file](https://github.com/fog/fog-brightbox/blob/master/lib/fog/brightbox.rb) which will load everything.

I'm closing this and will open a fresh new PR asap. (just for the relatives and autoload stuff)

Thanks for taking the time to review it!
